server/issue.go

```diff
@@ -976,6 +1062,88 @@ func getUpdateTask(database *api.Database, migrationType db.MigrationType, vcsPu
+		payload.SchemaVersion = schemaVersion
+		if vcsPushEvent != nil {
+			payload.VCSPushEvent = vcsPushEvent
+		}
+		bytes, err := json.Marshal(payload)
+		if err != nil {
+			return nil, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("failed to marshal database schema update gh-ost sync payload, error: %v", err))
+		}
+		taskCreateList = append(taskCreateList, api.TaskCreate{
+			Name:              fmt.Sprintf("Update %q schema gh-ost sync", database.Name),
+			InstanceID:        database.InstanceID,
+			DatabaseID:        &database.ID,
+			Status:            taskStatus,
+			Type:              api.TaskDatabaseSchemaUpdateGhostSync,
+			Statement:         d.Statement,
+			EarliestAllowedTs: d.EarliestAllowedTs,
+			MigrationType:     db.Migrate,
+			Payload:           string(bytes),
+		})
+	}
+	{
+		payload := api.TaskDatabaseSchemaUpdateGhostCutoverPayload{}
+		bytes, err := json.Marshal(payload)
+		if err != nil {
+			return nil, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("failed to marshal database schema update ghost cutover payload, error: %v", err))
+		}
+		taskCreateList = append(taskCreateList, api.TaskCreate{
+			Name:              fmt.Sprintf("Update %q schema gh-ost cutover", database.Name),
+			InstanceID:        database.InstanceID,
+			DatabaseID:        &database.ID,
+			Status:            taskStatus,
+			Type:              api.TaskDatabaseSchemaUpdateGhostCutover,
+			EarliestAllowedTs: d.EarliestAllowedTs,
+			Payload:           string(bytes),
+		})
+	}
+	{
+		parser := ghostsql.NewParserFromAlterStatement(d.Statement)
+		tableName := ""
+		if parser.HasExplicitTable() {
+			tableName = fmt.Sprintf("_%v_del", parser.GetExplicitTable())
+		}
+		payload := api.TaskDatabaseSchemaUpdateGhostDropOriginalTablePayload{
+			DatabaseName: database.Name,
+			TableName:    tableName,
+		}
+		bytes, err := json.Marshal(payload)
+		if err != nil {
+			return nil, nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("failed to marshal database schema update ghost drop original table payload, error: %v", err))
+		}
+		taskCreateList = append(taskCreateList, api.TaskCreate{
+			Name:              "Update %q schema gh-ost drop original table",
+			InstanceID:        database.Instance.ID,
+			DatabaseID:        &database.ID,
+			Status:            taskStatus,
+			Type:              api.TaskDatabaseSchemaUpdateGhostDropOriginalTable,
+			EarliestAllowedTs: d.EarliestAllowedTs,
+			Payload:           string(bytes),
+		})
+	}
+	// ID is actually the taskCreateList array index here because we don't know task ID before inserting.
+	// So the below list means that taskCreateList[0] blocks taskCreateList[1], taskCreateList[1] blocks taskCreateList[2], which makes sense because the "sync" task blocks the "cutover" task, and the "drop original table" task can only be done after "cutover".
+	taskDAGCreateList := []api.TaskDAGCreate{
+		{
+			FromTaskID: 0,
+			ToTaskID:   1,
```

Comment:
This looks a bit strange.. Should we have a separate one called ToTaskIDIndex instead?

Reply:
Do you mean that using `TaskDAGCreate` here is confusing because it's actually index here, so we should use change `taskDAGCreateList` to `ToTaskIDIndex`?

Reply:
ToTaskIDIndex sounds good.